docs/disclosure-log.md

```diff
@@ -94,3 +96,253 @@ This action does not enable because the light client on `B` has a more recent co
 Chain `B` drops this message.
 
 From this point on, the model stutters, i.e., is unable to progress further in the connection handshake protocol.
+
+
+### 3. ICS3 problems due to version negotiation
+
+__Context__.
+The original issue triggering this discussion is here: [cosmos/ics/#459](https://github.com/cosmos/ics/issues/459).
+Briefly, version negotiation in the ICS3 handshake can interfere in various ways, breaking either the safety or liveness of this protocol.
+Several solution candidates exist, which we classify by their "mode", i.e., a strategy for picking the version at some point or another in the protocol.
+For a full description of the modes, please consult [L2-tla/readme.md#version-negotiation-modes](L2-tla/README.md#version-negotiation-modes).
+
+__Overview__.
+Below we use TLA+ traces to explore and report on the exact problems that can occur. We also show how the solution candidates fare.
+These are the takeaways from the discussion below:
+
+1. The set of compatible versions that chains start off with (return values of `getCompatibleVersions()` in ICS3) have to intersect, otherwise a liveness issue occurs. This assumption is independent of the version negotiation mode. We report this in __case (a)__ below.
+2. Modes "overwrite", "onTryNonDet", and "onAckNonDet" all result in breaking the handshake protocol. See __cases (b), (c), and (d)__ below for traces.
+3. The deterministic modes "onTryDet" and "onAckDet" pass model checking, so a solution should be chosen among these two candidates (see the original issue for follow-up on the solution).
+
+##### Case (a). Empty version intersection causes liveness issue
```

Comment:
Could you add a summary table with the different modes and check which have problems and what kind? I think it would be useful to give an overall view of the per mode issues and also a place where we see differences among modes.